http/util.go

```diff
@@ -71,31 +71,40 @@ var (
 		"sys/rekey/",
 	}
 
-	adjustRequest = func(c *vault.Core, r *http.Request) (*http.Request, int) {
-		u := r.URL
-		p := u.Path[3:]
+	validateNamespace = func(r *http.Request) int {
+		ns, err := namespace.FromContext(r.Context())
+		if err != nil {
+			return http.StatusBadRequest
+		}
+
+		fullURL, err := url.JoinPath(ns.Path, r.URL.Path[4:])
+		if err != nil {
+			return http.StatusBadRequest
+		}
 
-		nsHeaderPath := namespace.Canonicalize(r.Header.Get(consts.NamespaceHeaderName))
 		for _, api := range restrictedAPIs {
-			if strings.HasSuffix(p, api) && (p != "/"+api || nsHeaderPath != "") {
-				return r, http.StatusBadRequest
+			apiSysRawRouted := "sys/raw/" + api
+			// if path not equals the restricted api path or not equals restricted api path with "sys/raw/" prefix
+			namespaceUseOfRestrictedEndpoint := (fullURL != api && fullURL != apiSysRawRouted) && strings.HasSuffix(fullURL, api)
+			namespacePath, _, ok := strings.Cut(fullURL, api)
+
+			// exclude any occurences of possible restricted APIs paths when path dictates that the suffix is a policy name
+			if ok && strings.HasSuffix(namespacePath, "sys/policies/acl/") {
```

Comment:
```suggestion
			if ok && (strings.HasSuffix(namespacePath, "sys/policies/acl/") || strings.HasSuffix(namespacePath, "sys/policy/")) {
```

As there are two policy paths (latter is legacy, prior to introduction of EGP in Vault Enterprise).

Reply:
Really makes me think, if someone wants to take it on, we do something like the SSH or PKI backend tests and interrogate OpenAPI and make sure that all paths that work in the root namespace work in a namespace or have an explicit exclusion as to why it shouldn't.

Reply:
> As there are two policy paths (latter is legacy, prior to introduction of EGP in Vault Enterprise).
adjusted

> Really makes me think, if someone wants to take it on, we do something like the SSH or PKI backend tests and interrogate OpenAPI and make sure that all paths that work in the root namespace work in a namespace or have an explicit exclusion as to why it shouldn't.
Agree, still for future.